src/test/regress/expected/multi_router_planner.out

```diff
@@ -0,0 +1,1125 @@
+INSERT INTO articles_hash VALUES (37,  7, 'archduchies', 9997);
+INSERT INTO articles_hash VALUES (38,  8, 'anatine', 14067);
+INSERT INTO articles_hash VALUES (39,  9, 'anchises', 10906);
+INSERT INTO articles_hash VALUES (40, 10, 'attemper', 14976);
+INSERT INTO articles_hash VALUES (41,  1, 'aznavour', 11814);
+INSERT INTO articles_hash VALUES (42,  2, 'ausable', 15885);
+INSERT INTO articles_hash VALUES (43,  3, 'affixal', 12723);
+INSERT INTO articles_hash VALUES (44,  4, 'anteport', 16793);
+INSERT INTO articles_hash VALUES (45,  5, 'afrasia', 864);
+INSERT INTO articles_hash VALUES (46,  6, 'atlanta', 17702);
+INSERT INTO articles_hash VALUES (47,  7, 'abeyance', 1772);
+INSERT INTO articles_hash VALUES (48,  8, 'alkylic', 18610);
+INSERT INTO articles_hash VALUES (49,  9, 'anyone', 2681);
+INSERT INTO articles_hash VALUES (50, 10, 'anjanette', 19519);
+SET citus.task_executor_type TO 'real-time';
+SET citus.large_table_shard_count TO 2;
+SET client_min_messages TO 'DEBUG2';
+-- insert a single row for the test
+INSERT INTO articles_single_shard_hash VALUES (50, 10, 'anjanette', 19519);
+DEBUG:  Creating router plan
+DEBUG:  Plan is router executable
+-- first, test zero-shard SELECT, which should return an empty row
+SELECT COUNT(*) FROM articles_hash WHERE author_id = 1 AND author_id = 2;
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  predicate pruning for shardId 103300
+ count 
+-------
+      
+(1 row)
+
+-- single-shard tests
+-- test simple select for a single row
+SELECT * FROM articles_hash WHERE author_id = 10 AND id = 50;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  Plan is router executable
+ id | author_id |   title   | word_count 
+----+-----------+-----------+------------
+ 50 |        10 | anjanette |      19519
+(1 row)
+
+-- get all titles by a single author
+SELECT title FROM articles_hash WHERE author_id = 10;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  Plan is router executable
+   title    
+------------
+ aggrandize
+ absentness
+ andelee
+ attemper
+ anjanette
+(5 rows)
+
+-- try ordering them by word count
+SELECT title, word_count FROM articles_hash
+	WHERE author_id = 10
+	ORDER BY word_count DESC NULLS LAST;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  Plan is router executable
+   title    | word_count 
+------------+------------
+ anjanette  |      19519
+ aggrandize |      17277
+ attemper   |      14976
+ andelee    |       6363
+ absentness |       1820
+(5 rows)
+
+-- look at last two articles by an author
+SELECT title, id FROM articles_hash
+	WHERE author_id = 5
+	ORDER BY id
+	LIMIT 2;
+DEBUG:  Creating router plan
+DEBUG:  predicate pruning for shardId 103301
+DEBUG:  Plan is router executable
+  title  | id 
+---------+----
+ aruru   |  5
+ adversa | 15
+(2 rows)
+
+-- find all articles by two authors in same shard
```

Comment:
comment is not completely wrong but it missed a crucial point. There is an order by in the query. Therefore, even though this query hits a single shard, master node needs to sort the output afterward. Thus the plan is not router executable